### ISSUE TITLE
Pickling TypeError while serializing array of PyCtxt

### ISSUE THREAD

**Issue by albjerto** [ibarrond/Pyfhel#84]
I was following the code for encrypting numpy arrays in the repository examples, but the pickling step fails with a TypeError.

Code:
```python
import numpy as np
from Pyfhel import Pyfhel, PyCtxt

HE = Pyfhel()
HE.contextGen(p=65537)

HE.keyGen()

array1 = np.array([1.3,3.4,5.8,7.1,9.2])
array2 = np.array([-2., 4., -6., 8.,-10.])
arr_gen1 = np.empty(len(array1),dtype=PyCtxt)
arr_gen2 = np.empty(len(array1),dtype=PyCtxt)

for i in np.arange(len(array1)):
    arr_gen1[i] = HE.encryptFrac(array1[i])
    arr_gen2[i] = HE.encryptFrac(array2[i])

import pickle
arr1_pk = pickle.dumps(arr_gen1)
arr2_pk = pickle.dumps(arr_gen2)
pyfh_pk = pickle.dumps(HE)
```

Pickling fails at `arr1_pk = pickle.dumps(arr_gen1)`.
Traceback:
```
Traceback (most recent call last):
  File "D:\alberto\corsi\Tesi\venv\lib\site-packages\IPython\core\interactiveshell.py", line 3418, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-20-f2b2cc48d081>", line 51, in <module>
    arr1_pk = pickle.dumps(arr_gen1)
  File "stringsource", line 2, in Pyfhel.PyCtxt.PyCtxt.__reduce_cython__
TypeError: no default __reduce__ due to non-trivial __cinit__
```

Setup details:
 - Windows 10 
 - Python 3.8.6
 - C compiler MSC v.1927 64 bit (AMD64)
 - Pyfhel 2.2.5

**Comment by ibarrond** [comment]
This seems to be related to #81 , which was solved in commit bef4ce18fbc5ae77c4b58624c8e0c22b0821f41a (v.2.3.0). I guess you just need to update Pyfhel to the latest version and the issue should be gone.

**Comment by albjerto** [comment]
I can confirm this solved the issue, thank you.
